au4/src/au3wrap/internal/processinginteraction.cpp

```diff
@@ -60,3 +60,23 @@ bool ProcessingInteraction::changeClipTitle(const processing::ClipKey& clipKey,
 
     return true;
 }
+
+bool ProcessingInteraction::removeClip(const processing::ClipKey& clipKey, double begin, double end)
+{
+    WaveTrack* waveTrack = DomAccessor::findWaveTrack(projectRef(), TrackId(clipKey.trackId));
+    IF_ASSERT_FAILED(waveTrack) {
+        return false;
+    }
+
+    std::shared_ptr<WaveClip> clip = DomAccessor::findWaveClip(waveTrack, clipKey.index);
+    IF_ASSERT_FAILED(clip) {
+        return false;
+    }
+
+    clip->Clear(begin, end);
+
+    processing::ProcessingProjectPtr prj = globalContext()->currentProcessingProject();
+    prj->onClipChanged(DomConverter::clip(waveTrack, clip.get(), clipKey.index));
```

Comment:
I think it would be more correct to use onClipRemoved here to send a notification about the removed element in clipList